### PR TITLE
(almost) Fully working single player mode

### PR DESCRIPTION
**Pull request by gruszkat98**
Single player works like gold, however, it doesn't serialize properly, but that's not a major fix imo

### CODE REVIEW

File: src/edu/saddleback/tictactoe/controller/GameController.java [hunk 1]
```diff
@@ -138,7 +141,20 @@ public void setPlayer2Name(String player2Name) {
         this.player2Name = player2Name;
     }
 
-    public void setDifficulty(String difficulty) { this.difficultyString = difficulty; }
+    public void setDifficulty(String difficulty) {
```
Comment (typokign) [inline src/edu/saddleback/tictactoe/controller/GameController.java:140]: This should probably be an enum

File: src/edu/saddleback/tictactoe/decision/Minimax.java [hunk 2]
```diff
@@ -40,12 +42,35 @@ public int minimax(Node n, int depth, boolean max) {
         }
     }
 
+
+//    public static void main(String[] args) throws GridAlreadyChosenException {
```
Comment (typokign) [inline src/edu/saddleback/tictactoe/decision/Minimax.java]: cruft, delete please

File: src/edu/saddleback/tictactoe/decision/Node.java [hunk 3]
```diff
@@ -62,6 +62,39 @@ public static void generateTree(Node root){
 
     }
 
+//    public static Node findNode(Board board, Node node){
```
Comment (typokign) [inline src/edu/saddleback/tictactoe/decision/Node.java]: cruft

File: src/edu/saddleback/tictactoe/controller/GameController.java [hunk 4]
```diff
@@ -138,7 +141,20 @@ public void setPlayer2Name(String player2Name) {
...
+        if (difficulty == "Easy Mode")
+            MrBill.setEvaluator(new RandomEvaluator());
+        else
+            MrBill.setEvaluator(new AdvancedEvaluator());
+
```
Comment (typokign) [inline src/edu/saddleback/tictactoe/controller/GameController.java]: damn bro, you want some code with that whitespace?

**Review by typokign — DISMISSED** [review]
bill! bill! bill! bill!

**Review by kdcptkai31 — APPROVED** [review]

**Pull request merged**